roles/ceph-osd/tasks/check_devices.yml

```diff
@@ -5,7 +5,7 @@
 # As as a final word, I prefer to keep the partition check instead of running ceph-disk prepare with "failed_when: false"
 # I believe it's safer
 - name: check if the device is a partition or a disk
-  shell: "echo '{{ item }}' | egrep '/dev/(([a-z]{3,4}[0-9]$)|(cciss/c[0-9]{1}d[0-9]{1}p[0-9]$))'"
+  shell: "echo '{{ item }}' | egrep '/dev/[sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p][0-9]{1,2}$'"
```

Comment:
I think the first bracket is not needed here.
For me, this will do:

`/dev/sd[a-z]{1,2}$|hd[a-z]{1,2}$|cciss/c[0-9]{1}d[0-9]{1}p[0-9]$|nvme[0-9]n[0-9]p][0-9]{1,2}$`

What do you think?